### PR TITLE
DEVELOPER-1771 - Apply website theme to Drupal

### PR DESCRIPTION
**Pull request by LightGuard**

### CODE REVIEW

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
Previous failure was an intermittent Github error. Retrying

**Comment by ollyjshaw** [comment]
Previous failure was an intermittent google docs error. Retrying

**Comment by ollyjshaw** [comment]
retest this please

**Comment by paulrobinson** [comment]
needs a rebase

**Comment by paulrobinson** [comment]
@wesbos can you review this approach?

File: javascripts/namespace.js [hunk 1]
```diff
@@ -23,6 +19,10 @@ app.downloadManagerBaseUrl = "#{site.download_manager_base_url}";
 */
 app.cache = {};
 
+// tabzilla info
+app.tabzilla = {};
+app.tabzilla.middlewareLogo = "#{cdn(site.base_url + '/images/branding/RHJB_Middleware_Logotype.png')}";
```
Comment (paulrobinson) [inline javascripts/namespace.js]: Why do we still need this?

File: Gulpfile.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    '!javascripts/adobe-analytics.js'
+  ],
+  "styles": ['stylesheets/*.scss']
+};
+
+gulp.task('scripts',function() {
+  gulp.src(globs.scripts)
+    .pipe(plugins.plumber({errorHandler: plugins.notify.onError("Error: <%= error.message %>")}))
+    .pipe(plugins.concat('all.js'))
+    .pipe(plugins.injectString.wrap('(function($){', '})(jQuery);'))
```
Comment (wesbos) [inline Gulpfile.js]: What is this for? Does drupal not use `$` ?

Comment (LightGuard) [inline Gulpfile.js]: It loads it in compatibility mode, so I wrapped the whole javascript in function and passed in jQuery.

**Comment by paulrobinson** [comment]
Needs a rebase please.

**Comment by LightGuard** [comment]
Seriously?!

>

-- 
Sent from Gmail Mobile

**Comment by paulrobinson** [comment]
DCP 2 change went live today. That probably caused it.

**Comment by LightGuard** [comment]
I have to go through and make a few changes, it's not a simple merge it seems.

**Comment by paulrobinson** [comment]
Do I need to do anything special to run this? I did: `bundle exec ./control.rb -u --run-the-stack` and got:


Decrypting...
decrypting vault
Vault decrypted
Setting ports...
AWESTRUCT_HOST_PORT available at 32768
DRUPAL_HOST_PORT available at 32775
DRUPALPGSQL_HOST_PORT available at 32776
MYSQL_HOST_PORT available at 32777
SEARCHISKO_HOST_PORT available at 32778
Killing docker services...
args to docker compise are []
Stopping docker_drupal_1 ... done
Stopping docker_drupalpgsql_1 ... done
Stopping docker_searchisko_1 ... done
Stopping docker_mysql_1 ... done
Building...
Building CSS and JS for drupal
[12:43:33] Local gulp not found in ~/dev/developer.redhat.com
[12:43:33] Try running: npm install gulp
Error build CSS / JS for drupal

**Comment by LightGuard** [comment]
Yeah, see the bottom message. I forgot to add instructions to the readme.

On Monday, February 1, 2016, Paul Robinson <notifications@github.com> wrote:

> Do I need to do anything special to run this? I did: bundle exec
> ./control.rb -u --run-the-stack and got:
>
> Decrypting...
> decrypting vault
> Vault decrypted
> Setting ports...
> AWESTRUCT_HOST_PORT available at 32768
> DRUPAL_HOST_PORT available at 32775
> DRUPALPGSQL_HOST_PORT available at 32776
> MYSQL_HOST_PORT available at 32777
> SEARCHISKO_HOST_PORT available at 32778
> Killing docker services...
> args to docker compise are []
> Stopping docker_drupal_1 ... done
> Stopping docker_drupalpgsql_1 ... done
> Stopping docker_searchisko_1 ... done
> Stopping docker_mysql_1 ... done
> Building...
> Building CSS and JS for drupal
> [12:43:33] Local gulp not found in ~/dev/developer.redhat.com
> [12:43:33] Try running: npm install gulp
> Error build CSS / JS for drupal
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/redhat-developer/developers.redhat.com/pull/696#issuecomment-177960075>
> .
>


-- 
Sent from Gmail Mobile

**Comment by paulrobinson** [comment]
OK, I did:

npm install gulp
npm WARN package.json developers.redhat.com@1.0.0 No repository field.
npm WARN package.json developers.redhat.com@1.0.0 No license field.
npm WARN deprecated lodash@1.0.2: lodash@<2.0.0 is no longer maintained. Upgrade to lodash@^3.0.0
gulp@3.9.0 ../node_modules/gulp
├── interpret@0.6.6
├── deprecated@0.0.1
├── pretty-hrtime@1.0.1
├── archy@1.0.0
├── minimist@1.2.0
├── tildify@1.1.2 (os-homedir@1.0.1)
├── v8flags@2.0.11 (user-home@1.1.1)
├── semver@4.3.6
├── chalk@1.1.1 (supports-color@2.0.0, ansi-styles@2.1.0, escape-string-regexp@1.0.4, strip-ansi@3.0.0, has-ansi@2.0.0)
├── orchestrator@0.3.7 (stream-consume@0.1.0, sequencify@0.0.7, end-of-stream@0.1.5)
├── liftoff@2.2.0 (extend@2.0.1, rechoir@0.6.2, flagged-respawn@0.3.1, resolve@1.1.7, findup-sync@0.3.0)
├── vinyl-fs@0.3.14 (graceful-fs@3.0.8, strip-bom@1.0.0, vinyl@0.4.6, defaults@1.0.3, mkdirp@0.5.1, through2@0.6.5, glob-stream@3.1.18, glob-watcher@0.0.6)
└── gulp-util@3.0.7 (array-differ@1.0.0, array-uniq@1.0.2, lodash._reescape@3.0.0, lodash._reinterpolate@3.0.0, lodash._reevaluate@3.0.0, beeper@1.1.0, object-assign@3.0.0, fancy-log@1.1.0, replace-ext@0.0.1, has-gulplog@0.1.0, vinyl@0.5.3, gulplog@1.0.0, lodash.template@
3.6.2, through2@2.0.0, multipipe@0.1.2, dateformat@1.0.12)


And now I get:

bundle exec ./control.rb -u --run-the-stack
Decrypting...
decrypting vault
Vault decrypted
Setting ports...
AWESTRUCT_HOST_PORT available at 32768
DRUPAL_HOST_PORT available at 32769
DRUPALPGSQL_HOST_PORT available at 32770
MYSQL_HOST_PORT available at 32771
SEARCHISKO_HOST_PORT available at 32774
Killing docker services...
args to docker compise are []
Building...
Building CSS and JS for drupal
[13:57:55] Working directory changed to ~/dev/developer.redhat.com
module.js:338
    throw err;
          ^
Error: Cannot find module 'gulp-load-plugins'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/Users/paul/dev/developer.redhat.com/Gulpfile.js:2:15)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
Error build CSS / JS for drupal

**Comment by LightGuard** [comment]
`npm install` will install all dependencies.

On Monday, February 1, 2016, Paul Robinson <notifications@github.com> wrote:

> OK, I did:
>
> npm install gulp
> npm WARN package.json developers.redhat.com@1.0.0 No repository field.
> npm WARN package.json developers.redhat.com@1.0.0 No license field.
> npm WARN deprecated lodash@1.0.2: lodash@<2.0.0 is no longer maintained.
> Upgrade to lodash@^3.0.0
> gulp@3.9.0 ../node_modules/gulp
> ├── interpret@0.6.6
> ├── deprecated@0.0.1
> ├── pretty-hrtime@1.0.1
> ├── archy@1.0.0
> ├── minimist@1.2.0
> ├── tildify@1.1.2 (os-homedir@1.0.1)
> ├── v8flags@2.0.11 (user-home@1.1.1)
> ├── semver@4.3.6
> ├── chalk@1.1.1 (supports-color@2.0.0, ansi-styles@2.1.0,
> escape-string-regexp@1.0.4, strip-ansi@3.0.0, has-ansi@2.0.0)
> ├── orchestrator@0.3.7 (stream-consume@0.1.0, sequencify@0.0.7,
> end-of-stream@0.1.5)
> ├── liftoff@2.2.0 (extend@2.0.1, rechoir@0.6.2, flagged-respawn@0.3.1,
> resolve@1.1.7, findup-sync@0.3.0)
> ├── vinyl-fs@0.3.14 (graceful-fs@3.0.8, strip-bom@1.0.0, vinyl@0.4.6,
> defaults@1.0.3, mkdirp@0.5.1, through2@0.6.5, glob-stream@3.1.18,
> glob-watcher@0.0.6)
> └── gulp-util@3.0.7 (array-differ@1.0.0, array-uniq@1.0.2,
> lodash._reescape@3.0.0, lodash._reinterpolate@3.0.0,
> lodash._reevaluate@3.0.0, beeper@1.1.0, object-assign@3.0.0,
> fancy-log@1.1.0, replace-ext@0.0.1, has-gulplog@0.1.0, vinyl@0.5.3,
> gulplog@1.0.0, lodash.template@
> 3.6.2, through2@2.0.0, multipipe@0.1.2, dateformat@1.0.12)
>
> And now I get:
>
> bundle exec ./control.rb -u --run-the-stack
> Decrypting...
> decrypting vault
> Vault decrypted
> Setting ports...
> AWESTRUCT_HOST_PORT available at 32768
> DRUPAL_HOST_PORT available at 32769
> DRUPALPGSQL_HOST_PORT available at 32770
> MYSQL_HOST_PORT available at 32771
> SEARCHISKO_HOST_PORT available at 32774
> Killing docker services...
> args to docker compise are []
> Building...
> Building CSS and JS for drupal
> [13:57:55] Working directory changed to ~/dev/developer.redhat.com
> module.js:338
> throw err;
> ^
> Error: Cannot find module 'gulp-load-plugins'
> at Function.Module._resolveFilename (module.js:336:15)
> at Function.Module._load (module.js:278:25)
> at Module.require (module.js:365:17)
> at require (module.js:384:17)
> at Object. (/Users/paul/dev/developer.redhat.com/Gulpfile.js:2:15)
> at Module._compile (module.js:460:26)
> at Object.Module._extensions..js (module.js:478:10)
> at Module.load (module.js:355:32)
> at Function.Module._load (module.js:310:12)
> at Module.require (module.js:365:17)
> at require (module.js:384:17)
> Error build CSS / JS for drupal
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/redhat-developer/developers.redhat.com/pull/696#issuecomment-177983069>
> .
>


-- 
Sent from Gmail Mobile

**Comment by paulrobinson** [comment]
See: http://docker:32769/web-and-api-development/overview, notice the following JS error:

      Uncaught ReferenceError: jQuery is not defined

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by ollyjshaw** [comment]
retest this please

**Comment by paulrobinson** [comment]
This time, when I built it I see: https://www.dropbox.com/s/51g42a27a5dffal/Screenshot%202016-02-03%2013.21.03.png?dl=0

Let me know if you need to see any logs.

**Comment by LightGuard** [comment]
Looks like the theme didn't load

On Wednesday, February 3, 2016, Paul Robinson <notifications@github.com>
wrote:

> This time, when I built it I see:
> https://www.dropbox.com/s/51g42a27a5dffal/Screenshot%202016-02-03%2013.21.03.png?dl=0
>
> Let me know if you need to see any logs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/redhat-developer/developers.redhat.com/pull/696#issuecomment-179225684>
> .
>


-- 
Sent from Gmail Mobile

**Comment by LightGuard** [comment]
You'll have to look at the docker logs or the Drupal log to see what
happened.



-- 
Sent from Gmail Mobile

**Comment by paulrobinson** [comment]
I've emailed you the Drupal log, can you take a look?

**Comment by ollyjshaw** [comment]
docker build

**Comment by paulrobinson** [comment]
retest this please

**Comment by paulrobinson** [comment]
This seems to be working, when i built it on my laptop. I'll merge once the CI checks are done.

**Pull request merged**